internal/drivers/s3/s3.go

```diff
@@ -148,14 +165,26 @@ func (p *s3Driver) connect(ctx context.Context, urlString string) error {
 		o.UsePathStyle = true
 	})
 
+	if _, err := p.s3.ListObjects(ctx, &awss3.ListObjectsInput{Bucket: aws.String(p.bucket), MaxKeys: aws.Int32(1)}); err != nil {
+		var bnf *types.NoSuchBucket
+		if errors.As(err, &bnf) {
+			if _, err := p.s3.CreateBucket(ctx, &awss3.CreateBucketInput{Bucket: aws.String(p.bucket)}); err != nil {
```

Comment:
Just an outsiders perspective, but I would recommend **against** giving it the ability to create buckets unless there’s an “opt-in” flag or something to allow for more purposeful configuration. That could very easily lead to misconfigurations where sensitive data could end up in buckets that don’t have the appropriate security provisions on it. And if this tool is responsible for doing so, could that have liability implications?

Reply:
yes, i will consider that. however, by default most systems require you to explicitly given these elevated permissions.

Reply:
OK, @byrider-jparker4 good feedback. just changed it to only create when testing against a local s3 bucket (something like local stack).